tpdatasrc/tpgamefiles/scr/tpModifiers/sp_magic_circle_outward_fixed.py

```diff
@@ -0,0 +1,231 @@
+magicCircleOutward.AddHook(ET_OnD20Query, EK_Q_Critter_Has_Spell_Active, MagicCircleOutwardHasSpellActive, ())
+magicCircleOutward.AddHook(ET_OnD20Signal, EK_S_Killed, MagicCircleOutwardKilled, ())
+magicCircleOutward.AddHook(ET_OnD20Signal, EK_S_Spell_End, MagicCircleOutwardSpellEnd, ())
+magicCircleOutward.AddHook(ET_OnD20PythonQuery, "Has Magic Circle Spell", HasMagicCircleQuery, ())
+magicCircleOutward.AddHook(ET_OnGetTooltip, EK_NONE, MagicCircleOutwardTooltip, ())
+magicCircleOutward.AddHook(ET_OnGetEffectTooltip, EK_NONE, MagicCircleOutwardEffectTooltip, ())
+magicCircleOutward.AddSpellDispelCheckStandard()
+magicCircleOutward.AddSpellTeleportPrepareStandard()
+magicCircleOutward.AddSpellTeleportReconnectStandard()
+magicCircleOutward.AddSpellCountdownStandardHook()
+
+
+def MagicCircleOutwardEffAoEExited(attachee, args, evt_obj):
+	obj_evt_id = args.get_arg(0)
+	if obj_evt_id != evt_obj.evt_id:
+		print "Magic Circle Outward Aura: ID mismatch " + str(evt_obj.evt_id) + ", stored was: " + str(obj_evt_id)
+		return 0
+	print "Magic Circle Outward Aura (ID " + str(obj_evt_id) +") Exited, critter: " + attachee.description + " "
+	args.condition_remove()
+	return 0
+
+def MagicCircleOutwardEffTooltip(attachee, args, evt_obj):
+	type = args.get_arg(1)
+	TypeName = GetSpellType(type)
+	evt_obj.append("Protection from " + TypeName)
+	return 0
+	
+def MagicCircleOutwardEffEffectTooltip(attachee, args, evt_obj):
+	type = args.get_arg(1)
+	TypeName = GetSpellType(type)
+	evt_obj.append(tpdp.hash("MAGIC_CIRCLE_OUTWARD_FIXED_EFFECT"), -2, TypeName + "")
+	return 0
+
+def MagicCircleOutwardEffRemove(attachee, args, evt_obj):
+	print "Removing Magic Circle Effect Condition " + attachee.description
+	args.condition_remove()
+	return 0
+	
+def MagicCircleOutwardEffACBonus(attachee, args, evt_obj):
+	attacker = evt_obj.attack_packet.attacker
+	if attacker == OBJ_HANDLE_NULL:
+		return 0
+	
+	type = args.get_arg(1)
+	addBonus = CheckAlignment(attacker, type)
+	
+	if addBonus:
+		evt_obj.bonus_list.add(2, 11, 207) #Deflection bonus
+	return 0
+
+def MagicCircleOutwardEffSavingThrow(attachee, args, evt_obj):
+	caster = evt_obj.obj
+	type = args.get_arg(1)
+	
+	addBonus = CheckAlignment(caster, type)
+	if addBonus:
+		evt_obj.bonus_list.add(2, 15, 207) #Resistance bonus
+	return 0
+	
+def MagicCircleOutwardEffDamageResistance(attachee, args, evt_obj):
+	attacker = evt_obj.attack_packet.attacker
+	if attacker.d20_query_has_condition("sp-Summoned"):
+		type = args.get_arg(1)
+		canAttack = SummonCanAttack(attacker, type)
+		if not canAttack:
+			evt_obj.damage_packet.add_mod_factor(0.0, D20DT_UNSPECIFIED, 104) #Do no damage at all
+	return 0
+	
+def MagicCircleOutwardEffSpellImmunity(attachee, args, evt_obj):
+	sp_pkt = evt_obj.spell_packet
+	spell_enum = sp_pkt.spell_enum
+	if (spell_enum == 0):
+		return 0
+	
+	# Providing immunity (the effect really should just be suppressed while inside the circle)
+	if spell_enum in immunity_list:
```

Comment:
It should protect against a creature's natural attacks (circle really should keep summons out of the circle but I am trying to simplify it).  My understanding is that it is done incorrectly in prot vs evil though (see bug 165).  Should I check if the attacker has a weapon instead?

Reply:
I was talking about the spell suppression effect. Or was that unrelated?

Reply:
Sorry, I was thinking about a different issue when I replied (summons using weapons).  I can change the suppression to use the mental descriptor.

Reply:
Yes. For reference I already implemented it for prot vs evil, I think in the big spell immunity handler.
As for bug 165, strange, I don't remember it protecting vs Balor weapons... could be mosreported, or perhaps fixed in co8.

Reply:
I see the immunity code you are talking about.  I'll do it that way.  I would expect it to work correctly for the balor guardian but not a summoned balor.  I think the fix would be to check if the summon is attacking with a weapon.

Reply:
Sounds right.
Come to think of it, I think I've already hooked one of the magic circle funcs due to a ctd issue... make sure your code includes that.

Reply:
I'll look into the hook to make sure I don't add the same problem back.  I did a bit of research into the change you suggested (granting immunity to mind effecting).  I don't think that is correct now.  It would provide immunity to sleep and confusion (for example) which would be incorrect for prot vs alignment.  I think the way I have it in code now is the most correct way to do it (with the exception of adding a new mind control descriptor).